File: state.py
class State:
	def handle_event(self):
		print("event completed")
		
class Op1_state(State):
	def __init__(self, name):
		self.name = name
	def handle_event(self):
		print(self.name)
		
class Op2_state(State):
	def __init__(self, price):
		self.price = price
	def handle_event(self):
		print("op2")

File: State.java
abstract class State
{
	abstract public void handle_event();
}

class Op1State extends State
{
	private String name;
	private final EventType e = EventType.OP1;

	public Op1State(String name)
	{
		this.name = name;
	}

	public void handle_event()
	{
		System.out.println(name);
		Dispatch.notify1();
	}
}

class Op2State extends State
{
	private int price;
	private final EventType e = EventType.OP2;

	public Op2State(int price)
	{
		this.price = price;
	}

	public void handle_event()
	{
		System.out.println(price);
		Dispatch.notify1();
	}
}